docs/guides/latest/workloads/deployment.md

```diff
@@ -0,0 +1,435 @@
+
+**Create BackupConfiguration:**
+
+Below is the YAML of the `BackupConfiguration` crd that we are going to create,
+
+```console
+apiVersion: stash.appscode.com/v1beta1
+kind: BackupConfiguration
+metadata:
+  name: deployment-backup
+  namespace: demo
+spec:
+  repository:
+    name: local-repo
+  schedule: "*/1 * * * *"
+  target:
+    ref:
+      apiVersion: apps/v1
+      kind: Deployment
+      name: stash-demo
+    volumeMounts:
+    - name: source-data
+      mountPath: /source/data
+    directories:
+    - /source/data
+  retentionPolicy:
+    name: 'keep-last-5'
+    keepLast: 5
+    prune: true
+```
+
+Here,
+
+- `spec.repository` refers to the `laocal-repo` local Backend.
+- `spec.schedule` is a cron expression indicates that `BackupSession` will be created at 1 minute interval.
+- `spec.target.ref` refers to the target workload that was created for `stash-demo` Deploymnet.
+
+Let's create the `BackupConfiguration` crd we have shown above,
+
+```console
+ $ kubectl apply -f ./backupconfiguration.yaml
+backupconfiguration.stash.appscode.com/deployment-backup created
+```
+
+**Verify CronJob:**
+
+If everything goes well, Stash will create a `CronJob` with the schedule specified in `spec.schedule` field of `BackupConfiguration` crd.
+
+Verify that the `CronJob` has been created using the following command,
+
+```console
+$ kubectl get cronjob -n demo
+NAME                SCHEDULE      SUSPEND   ACTIVE   LAST SCHEDULE   AGE
+deployment-backup   */1 * * * *   False     0        35s             64s
+```
+
+**Wait for BackupSession:**
+
+The `deployment-backup` CronJob will trigger a backup on each schedule by creating a `BackpSession` crd.
+
+Wait for a schedule to appear. Run the following command to watch `BackupSession` crd,
+
+```console
+$ watch -n 2 kubectl get backupsession -n demo 
+Every 1.0s: kubectl get backupsession -n demo     suaas-appscode: Mon Jun 24 10:23:08 2019
+
+NAME                           BACKUPCONFIGURATION   PHASE       AGE
+deployment-backup-1561350065   deployment-backup     Running     30s
+deployment-backup-1561350125   deployment-backup     Succeeded   63s
+```
+
+We can see above that the backup session has succeeded. Now, we will verify that the backed up data has been stored in the local backend.
+
+**Verify Backup:**
+
+Once a backup is complete, Stash will update the respective `Repository` crd to reflect the backup. Check that the repository `local-repo` has been updated by the following command,
+
+```console
+ $ kubectl get repository -n demo local-repo
+NAME         INTEGRITY   SIZE   SNAPSHOT-COUNT   LAST-SUCCESSFUL-BACKUP   AGE
+local-repo   true        0 B    1                27s                      1m4s
+```
+
+## Restore Deployment's Volumes
+
+This section will show you how to restore the backed up data from the backend we have taken in earlier section.
+
+**Deploy Deployment:**
+
+Now, we have to deploy a new Deployment whether the backed up data has been restored.
```

Comment:
We are going to create a new Deployment named `stash-recovered` and restore the inside it.